### ISSUE TITLE
Can't move mocks

### ISSUE THREAD

**Issue by j-mok** [eranpeer/FakeIt#97]
Mocks seem to poorly handle moving and copying, causing crashes. It would be convenient to be able to write helper functions that construct mocks, then stub them as needed and move them back to the caller.

**Comment by vijairaj** [comment]
Like @j-mok  said, it would be convenient to have helper methods for constructing and returning mocks but fakeit behaviour seems to be inconsistent. I have given a reduced test case to reproduce the behaviour with Visual studio 2017.

```c++
struct IAppliance {
    ~IAppliance() = default;
    virtual void SwitchOn() = 0;
    virtual void SwitchOff() = 0;
};

template<typename T>
fakeit::Mock<T> CreateFakeAppliance()
{
    using namespace fakeit;

    Mock<T> mock;
    When(Method(mock, SwitchOn)).AlwaysReturn();
    When(Method(mock, SwitchOff)).AlwaysReturn();
    return mock;
}

int main()
{
    auto mock = CreateFakeAppliance<IAppliance>();

    // cl /MD /EHsc /O2 test.cpp - works
    // cl /MD /EHsc test.cpp - crashes in the below line (not the missing /O2 flag)
    mock.get().SwitchOn();

    return 0;
}

```

**Comment by domainerror** [comment]
I've run into this, too. I think it would make sense to prevent copying and moving mocks until this issue is addressed.

**Comment by FranckRJ** [comment]
Fixed by #334, now mocks are movable (they can't be copyable as explained in the PR). It will be released in the next version of FakeIt.
